internal/character/march7th/eidolon.go

```diff
@@ -0,0 +1,82 @@
+package march7th
+
+import (
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/engine/modifier"
+	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
+)
+
+const (
+	MarchE1       = "march7th-e1"
+	MarchE2       = "march7th-e2"
+	MarchE2Shield = "march7th-e2-shield"
+)
+
+func init() {
+	modifier.Register(MarchE2Shield, modifier.Config{
+		Listeners: modifier.Listeners{
+			OnAdd: func(mod *modifier.Instance) {
+				mod.Engine().AddShield(MarchE2Shield, info.Shield{
+					Source: mod.Source(),
+					Target: mod.Owner(),
+					BaseShield: info.ShieldMap{
+						model.ShieldFormula_SHIELD_BY_SHIELDER_DEF: 0.24,
+					},
+					ShieldValue: 320,
+				})
+			},
+			OnRemove: func(mod *modifier.Instance) {
+				mod.Engine().RemoveShield(MarchE2Shield, mod.Owner())
+			},
+		},
+	})
+
+	modifier.Register(MarchE2, modifier.Config{
+		Stacking: modifier.Replace,
+		Listeners: modifier.Listeners{
+			OnAdd: determineE2Target,
+		},
+	})
```

Comment:
This does not do what you want. `OnAdd` just runs when the modifier is added. How you are using it will just run immediately when `AddModifier` is called.

Instead of using a modifier for this, should just use a `BattleStart` subscription to defer the execution until when the battel starts (since need to wait for all characters to be created first)